src/client/scripts/game/chess/checkdetection.js

```diff
@@ -230,10 +241,20 @@ const checkdetection = (function(){
         return movepiece.simulateMove(gamefile, move, color).isCheck;
     }
 
-    // Time complexity: O(1)
+    // Time complexity: O(slides) basically O(1) unless you add a ton of slides to a single piece
     function removeSlidingMovesThatPutYouInCheck (gamefile, moves, pieceSelected, color) {
+        // O(1)
+        function isEmpty(obj) {
+            for (var prop in obj) {
+                if (Object.prototype.hasOwnProperty.call(obj, prop)) {
+                    return false;
+                }
+            }
+          
+            return true
+        }
 
```

Comment:
This function would be good to place in `math`

Reply:
isEmpty has been moved to math